### ISSUE TITLE
Check for type params of more classes in akka-serializability-plugin

### ISSUE THREAD

**Issue by PawelLipski** [VirtusLab/akka-safer-serializer#36]
Not sure what else (other than what's already in `org.virtuslab.akkasaferserializer.SaferSerializerPluginComponent#newPhase` -> `genericsNames`) to include, but I guess this can be checked systematically (reflection?) by going over type params of classes in akka and finding out which of them correspond to message/event/state types.

**Comment by MarconZet** [comment]
Do you mean like creating a custom, one-time usage plugin that would go over Hydra and collect references by generic arguments of `@SerializabilityTrait`?

**Comment by PawelLipski** [comment]
Ideally, I'd rather go over all(-ish) classes in common Akka dependencies and see which of them **have type params** corresponding to messages/events/persistent state :goncern:

**Comment by PawelLipski** [comment]
I know this might be hard to achieve tho, unless Akka is very rigorous about always naming type params in an unambiguous way; good news is that probably won't require any plugin, just basic reflection should be enough

**Comment by PawelLipski** [comment]
I think it doesn't even need to a **one-time** tool, it can just live in this repository, to be used in future (maybe with some slight modifications if needed)

**Comment by MarconZet** [comment]
> Ideally, I'd rather go over all(-ish) classes in common Akka dependencies and see which of them **have type params** corresponding to messages/events/persistent state :goncern:

I don't understand.
Type parameter is just a generic, nothing inherently links it to serialisation.

**Comment by PawelLipski** [comment]
> Type parameter is just a generic, nothing inherently links it to serialisation.

Yes, I agree that nothing about the type param links it serialization.
However, our purpose here is **to detect the set of user-defined types that are messages, events and persistent state** as accurately as possible.

AFAIK the reasonable way to do that is to see which types has been used as type parameters of types like `akka.actor.typed.Behavior`, `akka.persistence.typed.scaladsl.Effect`, which we already do... but also e.g. `akka.actor.typed.ActorRef` (which we don't so far :/).
In general, it's non-trivial to collect the entire such set of Akka-defined types... the clue that we can is the fact that in certain cases, these type params have explicit names, e.g.:

```scala
trait ReplyEffect[+Event, State] extends Effect[Event, State]
```

Although relying on type param names wouldn't work e.g. for `akka.actor.typed.ActorRef`:

```scala
trait ActorRef[-T] extends RecipientRef[T] with java.lang.Comparable[ActorRef[_]] with java.io.Serializable
```
